src/Snowflake/Command.php

```diff
@@ -149,18 +226,270 @@ public static function cloneDatabaseFromShared(Connection $connection, array $da
+    {
+        $connection->query(sprintf(
+            'GRANT ROLE %s TO USER %s',
+            QueryBuilder::quoteIdentifier($role),
+            QueryBuilder::quoteIdentifier($user)
+        ));
+    }
+
+    public static function assignGrantToRole(Connection $connection, array $grant): void
+    {
+        self::useRole($connection, $grant['granted_by']);
+
+        if ($grant['privilege'] === 'USAGE' && $grant['granted_on'] === 'ROLE') {
+            $sql = sprintf(
+                'GRANT %s %s TO %s %s %s',
+                $grant['granted_on'],
+                $grant['name'],
+                $grant['granted_to'],
+                $grant['grantee_name'],
+                $grant['grant_option'] === 'true' ? 'WITH GRANT OPTION' : '',
+            );
+        } else {
+            $sql = sprintf(
+                'GRANT %s ON %s %s TO %s %s %s',
+                $grant['privilege'],
+                $grant['granted_on'],
+                $grant['granted_on'] !== 'ACCOUNT' ? $grant['name'] : '',
+                $grant['granted_to'],
+                $grant['grantee_name'],
+                $grant['grant_option'] === 'true' ? 'WITH GRANT OPTION' : '',
+            );
+        }
+
+        $connection->query($sql);
+    }
+
+    public static function createMainRole(
+        Connection $connection,
+        string $warehouse,
+        array $users
+    ): void {
+        $mainRole = 'KEBOOLA_STORAGE';
+        $user = $mainRole;
+
+        self::createRole($connection, ['name' => $mainRole]);
+
+        $connection->query(sprintf(
+            'DROP WAREHOUSE IF EXISTS %s',
+            $warehouse
+        ));
+
+        $connection->query(sprintf(
+            'GRANT CREATE DATABASE ON ACCOUNT TO ROLE %s;',
+            $mainRole
+        ));
+
+        $connection->query(sprintf(
+            'GRANT CREATE ROLE ON ACCOUNT TO ROLE %s WITH GRANT OPTION;',
+            $mainRole
+        ));
+
+        $connection->query(sprintf(
+            'GRANT CREATE USER ON ACCOUNT TO ROLE %s WITH GRANT OPTION;',
+            $mainRole
+        ));
+
+        $createWarehouseSql = <<<SQL
+'CREATE WAREHOUSE %s WITH WAREHOUSE_SIZE = 'XSMALL' WAREHOUSE_TYPE = 'STANDARD' AUTO_SUSPEND = 300 AUTO_RESUME = TRUE;'
+SQL;
+
+        $connection->query(sprintf(
+            $createWarehouseSql,
+            $warehouse
+        ));
+
+        $connection->query(sprintf(
+            'GRANT USAGE ON WAREHOUSE %s TO ROLE %s WITH GRANT OPTION;',
+            $warehouse,
+            $mainRole
+        ));
+
+        $connection->query(sprintf(
+            'USE WAREHOUSE %s',
+            $warehouse
+        ));
+
+        $connection->query(sprintf(
+            'DROP USER IF EXISTS %s',
+            $user
+        ));
+
+        $connection->query(sprintf(
+            'CREATE USER %s PASSWORD=%s DEFAULT_ROLE=%s',
+            $user,
+            QueryBuilder::quoteIdentifier($users[$user]),
```

Comment:
Nerozumím - tipoval bych, že bys měl heslo escapovat jako hodnotu, ne jako identifikator, ne? 

objectProperties ::=
  PASSWORD = '<string>'
  
  QuoteIdentifier ti vrátí `"identifier"`

Reply:
no však - převzaté z
https://keboolamanagementapi.docs.apiary.io/#reference/super-storage-backends-management/storage-backend-collection

Reply:
heslo i po založení usere funguje

Reply:
Aha, tak "if it works it ain't stupid" :)